src/solver/impls/slepc/slepc.cxx

```diff
@@ -23,202 +23,209 @@
-                                      PetscScalar *eigi){
+// The callback function for transforming the eigenvalues in the
+// custom shell spectral transformation
+PetscErrorCode stBackTransformWrapper(ST st, PetscInt nEig, PetscScalar* eigr,
+                                      PetscScalar* eigi) {
   PetscFunctionBegin;
-  //First get the context of the st object and cast to correct type
+  // First get the context of the st object and cast to correct type
   SlepcSolver* myCtx;
-  STShellGetContext(st,(void **)&myCtx);
+  STShellGetContext(st, (void**)&myCtx);
 
-  //Convert to bout eigenvalue
+  // Convert to bout eigenvalue
   BoutReal tmpR, tmpI;
-  for(PetscInt iEig=0;iEig<nEig;iEig++){
-    myCtx->slepcToBout(eigr[iEig],eigi[iEig],tmpR,tmpI,true);
-    eigr[iEig]=tmpR; eigi[iEig]=tmpI;
+  for (PetscInt iEig = 0; iEig < nEig; iEig++) {
+    myCtx->slepcToBout(eigr[iEig], eigi[iEig], tmpR, tmpI, true);
+    eigr[iEig] = tmpR;
+    eigi[iEig] = tmpI;
   };
   PetscFunctionReturn(0);
 }
 
+// Helper function
+std::string formatEig(BoutReal reEig, BoutReal imEig) {
 
-//Helper function
-std::string formatEig(BoutReal reEig, BoutReal imEig){
-  std::string rePad, imPad;
-
-  if(reEig<0){
-    rePad="-";
-  }else{
-    rePad=" ";
-  }
-
-  if(imEig<0){
-    imPad="-";
-  }else{
-    imPad="+";
-  }
+  const std::string rePad = (reEig < 0) ? "-" : " ";
+  const std::string imPad = (imEig < 0) ? "-" : "+";
 
   std::stringstream tmp;
-  tmp.precision(5); tmp<<std::scientific;
-  //Note we use abs here and put the -/+ into pads to cope with
-  //the case where the Eig is -0.000/0.000 which require different
-  //padding but evaluate as the same number when doing comparisons
-  tmp<<rePad<<std::abs(reEig)<<imPad<<std::abs(imEig)<<"i";
+  tmp.precision(5);
+  tmp << std::scientific;
+  // Note we use abs here and put the -/+ into pads to cope with
+  // the case where the Eig is -0.000/0.000 which require different
+  // padding but evaluate as the same number when doing comparisons
+  tmp << rePad << std::abs(reEig) << imPad << std::abs(imEig) << "i";
   return tmp.str();
 }
 
-SlepcSolver::SlepcSolver(Options *options){
+SlepcSolver::SlepcSolver(Options* options) {
   has_constraints = false;
   initialised = false;
-  stIsShell=PETSC_FALSE;
+  stIsShell = PETSC_FALSE;
 
   // Slepc settings in the Solver section
+  auto& options_ref = *options;
 
-  options->get("nEig",nEig,0);//0 means keep the current value, i.e. autoset
+  nEig = options_ref["nEig"]
+             .doc("Number of eigenvalues to compute. 0 means keep the current value, "
+                  "i.e. autoset")
+             .withDefault(0);
 
-  options->get("tol",tol,1.0e-6);// Tolerance --> Note this is on SLEPc eig not BOUT
-  options->get("maxIt",maxIt,PETSC_DECIDE);
+  tol = options_ref["tol"].doc("SLEPc tolerance").withDefault(1.0e-6);
+  maxIt = options_ref["maxIt"].doc("Maximum iterations").withDefault(PETSC_DEFAULT);
```

Comment:
I notice this has changed from `PETSC_DECIDE` to `PETSC_DEFAULT` is this intentional (and backwards compatible)?

Reply:
Yes -- this is what the docs say to use now, although in practice either is acceptable (had to check the source code to actually find out). Backwards compatible to 2006, which is much earlier than we support.